apps/homepage/src/app/[lang]/_shared/header/header.tsx

```diff
@@ -14,17 +14,25 @@ type HeaderProps = {
 };
 
 export function Header({ loc }: HeaderProps) {
+  const pathname = usePathname();
+  
   return (
     <>
       <UiViewport criteria="m|l|xl">
         <div className={styles.main} data-testid="header">
           <h1>@1<span className={styles.spanText}>x</span>Dev</h1>
           <div className={styles.buttonMenu}>
-            {buttons.map((button, index) => (
-              <IconLink key={index} href={button.href} icon={button.icon as UiIconProps["icon"]}>
-                {loc[button.key]}
-              </IconLink>
-            ))}
+            {buttons.map((button, index) => {
+              const isActive = 
+                (button.href === "/" && (pathname === "/en" || pathname === "/es")) || 
+                pathname === (`/en${button.href}`) || 
+                pathname === (`/es${button.href}`);
```

Comment:
Is there a way to simplify this? What if you break down the string and just get the part after the `/en` or `/es` to check the `href`?